src/Workspaces/Core/Portable/Utilities/AbstractSpeculationAnalyzer.cs

```diff
@@ -449,6 +449,15 @@ protected bool ReplacementChangesSemantics(SyntaxNode currentOriginalNode, Synta
             return false;
         }
 
+        protected bool ReplacementChangesSemanticsOfWhenNotNull(TExpressionSyntax originalWhenNotNull, TExpressionSyntax newWhenNotNull)
+        {
+            // We want to look at the first symbol following the `?.` and determine whether the symbols are compatible
+            var originalFirstNode = originalWhenNotNull.GetFirstToken().Parent;
```

Comment:
I had to look at the [language spec](https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#null-conditional-operator) to figure out the kind of expressions that can be legally present on the right side of the conditional access, and why your implementation is safe for all allowed cases. Consider adding a comment about the same.